SocialCareCaseViewerApi/V1/Gateways/DatabaseGateway.cs

```diff
@@ -68,6 +68,14 @@ public List<Allocation> SelectAllocations(string mosaicId)
             return allocations;
         }
 
+        public CreateAllocationRequest CreateAllocation(CreateAllocationRequest request)
```

Comment:
Return type of this method should be something else than the request object type. Return object should reflect the result of the performed action. Often in db methods you return the result of the db action, in this case the return value .SaveChanges() method.